andela/artemis-ah-backend
# #164198490 validate user registration details

#### What does this PR do?
Ensure that the appropriate error responses are displayed during user registration

#### Description of Task to be completed?
- Write unit tests
- Return a 400 and appropriate error message if email does not exist
- Return a 400 and appropriate error message if email is invalid
- Return a 400 and appropriate error message if password is not alphanumeric
- Return a 400 and appropriate error message if password is less than 8 characters
- Return a 409 if username or email already exists

#### How should this be manually tested?
1. Clone this branch.
2. Open up Postman
3. Make a POST request to the endpoint `/api/users` omitting the required fields or invalid request eg
of a malformed request body
```
{
    "email": "johndoejieu#$",
    "password": "12345678",
    "firstname": "John",
    "lastname": "Doe"
}
```
POST-ing the above request should return the following response
```
{
    "errors": {
        "email": [
            "Email is invalid."
        ]
    }
}
```

#### What are the relevant pivotal tracker stories?
[#164198490](https://pivotaltracker.com/story/show/164198490)
